### PR TITLE
Checkout: Point 'Set Up eCommerce' link to Calypso instead of wp-admin

### LINKED ISSUE #1518: Post Checkout: link to Business Plugins goes to wp-admin
In the Thank You screen, the "Set Up eCommerce" button takes you to wp-admin. It should take you to `/plugins` instead.

screenshot-wpcalypso wordpress com 2015-12-11 16-02-49

cc @drewblaisdell

### PR DESCRIPTION
**Pull request by drewblaisdell**
screenshot-wpcalypso wordpress com 2015-12-11 16-02-49

Fixes #1518.

This PR also contains some slight refactoring of `CheckoutThankYou`:

- Removes `PurchaseDetailButton` entirely, in favor of `Button`.
- Moves `PurchaseDetail` into its own component.

That file could use more refactoring - I believe it's one of the older modules in Calypso - but that is out-of-scope for this PR.

**Testing**
- Visit `/plans/:site` and purchase the business plan.
- Assert that clicking 'Set Up eCommerce' takes you to `/plugins/:site` in a new window instead of a `/wp-admin` route.

- [x] Product review
- [x] Code review

### CODE REVIEW

**Comment by drewblaisdell** [comment]
I'm going to update cb10f69d281a8517df6244ed9e8326bbcf72ab05 to refactor the CSS for that component as well.

File: client/my-sites/upgrades/checkout/purchase-detail.jsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ * External dependencies
+ */
+import React from 'react';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+
+const PurchaseDetail = ( { additionalClass, buttonText, description, onButtonClick, title } ) => (
```
Comment (stephanethomas) [inline client/my-sites/upgrades/checkout/purchase-detail.jsx]: Shouldn't we define `propTypes` as well?

Comment (drewblaisdell) [inline client/my-sites/upgrades/checkout/purchase-detail.jsx]: Good catch - I updated the second commit to assign `propTypes` for this component.

**Comment by drewblaisdell** [comment]
I updated the second commit to move the `PurchaseDetail` styles to `my-sites/upgrades/checkout/style.scss`. There is additional refactoring that can be done with that file, like switching to Gridicons, but that is unrelated to this issue and can be added separately.

**Comment by stephanethomas** [comment]
Looks good to me and works fine.

**Comment by fabianapsimoes** [comment]
> Assert that clicking 'Set Up eCommerce' takes you to /plugins/:site in a new window instead of a /wp-admin route.

Since this is now linking to Calypso, we can probably open within the same tab instead of using a new window.

**Comment by stephanethomas** [comment]
I think right now all links on the `Thank You` page open in a new window.

**Comment by drewblaisdell** [comment]
Yeah, I assumed that the behavior was intentional because it mirrors the behavior of the other internal links on that page. I'm happy to switch all of them though - it does seem like an improvement.

**Comment by drewblaisdell** [comment]
Actually, after working on that problem, I think it is something we should handle separately, as it requires changes to `<Button />`, needs to be tested with every product (including Jetpack plans), and this PR is already almost 200 lines of code. I opened an issue for opening these links internally here: #2810

That issue isn't a blocker for this (we already open other Calypso links in a new window on that page), so I'm going to put this back in review.

**Comment by drewblaisdell** [comment]
I added a solution for #2810 with #2816. This is now a blocker for that PR.

**Comment by fabianapsimoes** [comment]
Product :+1:

**Pull request merged**